### PR TITLE
feat: add contact archiving

### LINKED ISSUE #1623: Feature request: Contact archiving
**Problem:**
There are people I have long since fallen out of contact with cluttering up the contacts list.  However it would be nice to keep them around instead of deleting them entirely on the off chance I need to reach out to them again.

**Solution:**
Possibly add the ability to mark certain tags as hidden by default.  Or add something along the lines of the 'mark contacts as deceased' feature, though that doesn't hide them by default yet.

### PR DESCRIPTION
**Pull request by asbiin**
This close #1623 

- [x] Add "Archive"/"Unarchive" action on contact
image
- [x] Mask archived contacts from standard views
- [x] Add an archived contact view
- [x] Add a link to the archived contact view

(Add api functions to get archived contacts => this one will be done in another PR)

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -14,6 +14,7 @@ UNRELEASED CHANGES:
 * Fix gift recipient relation
 * Fix subscription cancel on account deletion
 * Fix email maximum size on settings
+* Add contact archiving
```
Comment (djaiss) [inline CHANGELOG]: `Add ability to archive a contact`

File: resources/assets/js/components/people/Archive.vue [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<style scoped>
+</style>
+
+<template>
+    <div>
+        <notifications group="archive" position="top middle" duration=5000 width="400" />
+
+        <a href="#" class="btn edit-information" @click="toggle">{{ active ? $t('people.contact_archive') : $t('people.contact_unarchive') }}</a>
```
Comment (djaiss) [inline resources/assets/js/components/people/Archive.vue]: I think you can simply omit `href="#"` entirely.

File: resources/views/people/_header.blade.php [hunk 3]
```diff
@@ -93,6 +93,9 @@
             <li>
               <a href="{{ route('people.edit', $contact) }}" class="btn edit-information" id="button-edit-contact">{{ trans('people.edit_contact_information') }}</a>
             </li>
+            <li>
+              <contact-archive hash="{{ $contact->hashID() }}" active="{{ json_encode($contact->is_active) }}"></contact-archive>
```
Comment (djaiss) [inline resources/views/people/_header.blade.php]: I don't think it should be put this feature here in the header. This is not a common action or even an important one. I would put it in the sidebar, just above the `Delete contact` option. You don't want people to easily access this feature, in my opinion.

File: resources/lang/en/people.php [hunk 4]
```diff
@@ -63,6 +63,7 @@
     'edit_contact_information' => 'Edit contact information',
     'contact_archive' => 'Archive contact',
     'contact_unarchive' => 'Unarchive contact',
+    'contact_archive_help' => 'Archived contact will not appear on contact lists',
```
Comment (djaiss) [inline resources/lang/en/people.php]: `Archived contacts will not be shown in the contact list`

**Comment by djaiss** [comment]
Don't forget to run Cypress on this one.

**Comment by asbiin** [comment]
@djaiss cypress fails on my computer on master branch: I cannot validates any modification with it.

**Comment by djaiss** [comment]
I've rewritten some of the screens, also removed the button and use a link instead.

**Pull request merged**